remote-vfs.c

```diff
@@ -0,0 +1,1889 @@
+
+	else if (!strcmp(arg, "scalar"))
+		rv_opts.param__mode = RV_MODE_SCALAR;
+
+	else if (!strcmp(arg, "none"))
+		rv_opts.param__mode = RV_MODE_NONE;
+
+	else
+		return error(_("invalid value for switch '%s'"),
+			     opt->long_name);
+
+	return 0;
+}
+
+static int option_parse_csu(const struct option *opt,
+			    const char *arg, int unset)
+{
+	if (unset) /* should not happen */
+		return error(_("missing value for switch '%s'"),
+			     opt->long_name);
+
+	else if (!strcmp(arg, "verify"))
+		rv_opts.param__csu = RV_CSU_VERIFY_DISABLE;
+
+	else if (!strcmp(arg, "error"))
+		rv_opts.param__csu = RV_CSU_VERIFY_ERROR;
+
+	else if (!strcmp(arg, "disable"))
+		rv_opts.param__csu = RV_CSU_DISABLE;
+
+	else if (!strcmp(arg, "allow"))
+		rv_opts.param__csu = RV_CSU_ALLOW;
+
+	else
+		return error(_("invalid value for switch '%s'"),
+			     opt->long_name);
+
+	return 0;
+}
+
+//////////////////////////////////////////////////////////////////
+
+/*
+ * The config is processed before we have parsed the command
+ * line args so we don't yet know which (if any) cache-server
+ * to use.  Remember both URLs and decide later.
+ *
+ * Also, look for well-known core config settings and capture
+ * values.  We do this directly so that we don't have to know
+ * about the existence of global variables from gvfs.c (or
+ * later, scalar.c).
+ *
+ * Also, break from the traditional config-helper model and
+ * do not intercept these values, but rather allow normal
+ * config processing continue.
+ */
+static int config_cb(const char *k, const char *v, void *data)
+{
+	if (!strcmp(k, "gvfs.cache-server"))
+		git_config_string(&rv_opts.param__vfs_url, k, v);
+
+	else if (!strcmp(k, "scalar.cache-server"))
+		git_config_string(&rv_opts.param__scalar_url, k, v);
+
+	else if (!strcmp(k, "core.gvfs"))
+		rv_opts.param__core_vfs = 1;
+
+	else if (!strcmp(k, "core.scalar"))
+		rv_opts.param__core_scalar = 1;
+
+	return git_default_config(k, v, data);
+}
+
+//////////////////////////////////////////////////////////////////
+
+/*
+ * Find the path to the ".gvfsCache" or ".scalarCache" ODB directory.
+ *
+ * The local ODB for this repo is in ".git/objects/".
+ *
+ * GVFS and Scalar create a shared object cache somewhere near the root of
+ * the drive.  This cache looks like a normal "alternate".
+ *
+ * Officially, the path to the shared cache is stored in the .gvfs folder
+ * (and outside of the repo).  It is available from the "gvfs status" command.
+ * Using it assumes our PATH is properly set up and we know which product's
+ * command to run.
+ */
+static void lookup_odb_paths(void)
+{
+	struct object_directory *odb;
+
+	prepare_alt_odb(the_repository);
+
+	/*
+	 * We now have a linked-list of ODBs.
+	 *
+	 * The first is always the local ODB (inside the .git directory).
+	 * Then come the alternates.  We should only have 1 alternate, but
+	 * can't complain if there are more.
+	 */
+
+	odb = the_repository->objects->odb;
+	if (!odb)
+		return;
```

Comment:
Should this be an error state? Or should we assign `rv_opts.param__*_odb_path = the_repository->objects->path`?

Reply:
Funny story.  Turns out that prepare_alt_obj() will segfault before we get here if there isn't an ODB.  The fix is to require the process to be inside a worktree.